hedera/api.py

```diff
@@ -310,19 +320,29 @@ def get_data(self):
             })
         return lang_list
 
+    def check_data(self, data):
+        keys = ["familiarity", "headword", "gloss", "vocabulary_list_id"]
+        for key in keys:
+            if key not in data:
+                return False
+        return True
+
     def post(self, request, *args, **kwargs):
         data = json.loads(request.body)
-        _, created = PersonalVocabularyListEntry.objects.get_or_create(**data)
-        return JsonResponse({"data": {"created": created}})
-
-# TODO add suggested node functionality
-# class LatticeNodesAPI(APIView):
-
-#     def get_data(self):
-#         headword = self.request.GET.get("headword")
-#         filtered_headword_iterable = filter(str.isalnum, headword)
-#         filtered_headword_string = "".join(filtered_headword_iterable)
-#         qs = LatticeNode.objects.filter(label__iregex=r"\y"+ re.escape(filtered_headword_string) + r"\y")
-#         data = serializers.serialize('json', qs)
-#         json_data = json.loads(data)
-#         return json_data
+        checked_data = self.check_data(data)
+        if checked_data is not True:
+            return HttpResponseBadRequest("Missing required fields")
```

Comment:
Should we return these 400 responses as JSON, that way the client knows the errors will be returned as JSON as well? Maybe something like this (just a suggestion):

```python
class JsonResponseBadRequest(JsonResponse):
    status_code = 400
```

And then you could write:

```python
return JsonResponseBadRequest(data={"error": "Missing required fields"})
```

Reply:
Yea, we definitely should! Will update the code to reflect this.